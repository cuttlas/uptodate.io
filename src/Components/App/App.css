.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  height: 15%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.App-grid{
  display: flex;
  flex-wrap:wrap;
  justify-content: center;
}
