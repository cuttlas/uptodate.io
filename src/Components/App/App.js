import React, { Component } from "react";
import { gql, graphql } from "react-apollo";
import InfiniteScroll from "react-infinite-scroller";
import styled from "styled-components";

import logo from "logo.svg";
import Article from "Components/Article/Article";

const query = gql`query getArticles ($cursor: String) {
  articles(first: 10, after: $cursor) {
    pageInfo {
      hasNextPage
      endCursor
    }
    edges {
      node {
        id
        title
        imgUrl
        url
        date
        description
        newsletters {
          id
          name
        }
      }
    }
  }
}`;

const Container = styled.div`
  text-align: center;
`;

const Logo = styled.img`
  height: 80px;
`;

const Header = styled.div`
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  background-color: white;
  height: 15%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const InfiniteGrid = styled(InfiniteScroll)`
  margin-top: 25px;
  display: flex;
  flex-wrap:wrap;
  justify-content: center;
`;

class App extends Component {
  static propTypes = {
    data: React.PropTypes.object
  };

  render() {
    const { articles, hasMore, fetchArticles } = this.props;

    return (
      <Container>
        <Header>
          <Logo src={logo} alt="logo" />
          <h2>WeWeekly</h2>
        </Header>
        <InfiniteGrid
          loadMore={fetchArticles}
          hasMore={hasMore}
          initialLoad={false}
        >
          {articles &&
            articles.map((article, key) => (
              <Article className="Article" article={article} key={key} />
            ))}
        </InfiniteGrid>
      </Container>
    );
  }
}

export { App };
export default graphql(query, {
  props({ data: { loading, articles, fetchMore } }) {
    return {
      articles: articles && articles.edges.map(edge => edge.node),
      hasMore: articles && articles.pageInfo.hasNextPage,
      loading,
      fetchArticles: () => {
        return fetchMore({
          query,
          variables: {
            cursor: articles.pageInfo.endCursor
          },
          updateQuery: (oldData, { fetchMoreResult }) => {
            const newData = fetchMoreResult.data;
            return {
              articles: {
                edges: [...oldData.articles.edges, ...newData.articles.edges],
                pageInfo: newData.articles.pageInfo
              }
            };
          }
        });
      }
    };
  }
})(App);
