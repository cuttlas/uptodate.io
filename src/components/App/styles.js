import styled from "styled-components";
import InfiniteScroll from "react-infinite-scroller";

export const Container = styled.div``;

export const InfiniteGrid = styled(InfiniteScroll)`
  margin-top: 5px;
`;
