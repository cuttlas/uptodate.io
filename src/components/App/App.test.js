import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { App } from "./App";

it("renders without crashing", () => {
  const props = {
    fetchArticles: jest.fn(),
    articles: [
      {
        title: "Mock Title",
        description: "Mock Description",
        url: "www.mock.com",
        imgUrl: "www.mock.com/mock.jpg"
      }
    ]
  };
  mount(<App articles={props.articles} fetchArticles={props.fetchArticles} />);
});

xit("renders the whole tree correctly", () => {
  const props = {
    fetchArticles: jest.fn(),
    articles: [
      {
        title: "Mock Title",
        description: "Mock Description",
        url: "www.mock.com",
        imgUrl: "www.mock.com/mock.jpg"
      }
    ]
  };
  const tree = renderer
    .create(
      <App articles={props.articles} fetchArticles={props.fetchArticles} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
