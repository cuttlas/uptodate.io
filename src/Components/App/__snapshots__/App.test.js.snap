exports[`test renders the whole tree correctly 1`] = `
<div
  className="sc-htpNat dEQUjC">
  <div
    className="sc-ifAKCX dsKbuV">
    <img
      alt="logo"
      className="sc-bxivhb dBqngZ"
      src="logo.svg" />
    <h2>
      WeWeekly
    </h2>
  </div>
  <div
    className="sc-EHOje jwKvwv">
    <a
      className="sc-bwzfXH dJuWjI"
      href="www.mock.com"
      target="_blank">
      <div
        className="sc-bdVaJa dcJUZD">
        <div
          className="Article-overlay iclOJA">
          <p
            className="Article-title leClUn">
            Mock Title
          </p>
          <p
            className="Article-description ggtoTs"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Mock Description",
              }
            } />
        </div>
      </div>
    </a>
  </div>
</div>
`;
